src/jit/hwintrinsicxarch.cpp

```diff
@@ -965,8 +965,7 @@ GenTree* Compiler::impSSEIntrinsic(NamedIntrinsic        intrinsic,
             assert(sig->numArgs == 1);
             assert(getBaseTypeOfSIMDType(sig->retTypeSigClass) == TYP_FLOAT);
             op1     = impPopStack().val;
-            retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, gtCloneExpr(op1), gtNewIconNode(0), NI_SSE_Shuffle,
-                                               TYP_FLOAT, simdSize);
+            retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, gtNewIconNode(0), NI_SSE_Shuffle, TYP_FLOAT, simdSize);
```

Comment:
Wouldn't this require more changes in codegen to ensure that the appropriate overload of `emitIns_SIMD` is called?

Reply:
Probably not, I saw the `Vector<T>` code always uses `shuffle` as 2-op form. Let me investigate more.

Reply:
It is not necessary to get new internal overload for `SSE_Shuffle` see comment: https://github.com/dotnet/coreclr/pull/16758#discussion_r172340325

You can substitute:
```C++
retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, gtNewIconNode(0), NI_SSE_Shuffle,
                                               TYP_INT, simdSize);
```

I think it's safe as I do not know any processors up to 10 years back which would support SSE and not support SSE2

Reply:
Otherwise, what I have seen in #16758 with `Compiler::fgMakeMultiUse` works really well with HA intrinsics.

Reply:
> Otherwise, what I have seen in #16758 with Compiler::fgMakeMultiUse works really well with HW intrinsics.

Why not adopt a simpler solution?

Reply:
I would just use `SSE2_Shuffle` we just have it already.

Reply:
I just found that this solution does not work because VEX-encoding always duplicates `dst` rather than `src` for this instruction.